string.c

```diff
@@ -11525,6 +11525,260 @@ rb_enc_interned_str_cstr(const char *ptr, rb_encoding *enc)
  *  without a ``!'' return a new String.  However, there are
  *  exceptions, such as String#[]=.
  *
+ *  == What's Here
+ *
+ *  First, what's elsewhere. \String includes the module Comparable,
+ *  which provides several very useful methods.
+ *
+ *  Here, class \String provides methods that are useful for:
+ *
+ *  - {Creating a String}[#class-String-label-Methods+for+Creating+a+String]
+ *  - {Frozen/Unfrozen Strings}[#class-String-label-Methods+for+a+Frozen-2FUnfrozen+String]
+ *  - {Querying}[#class-String-label-Methods+for+Querying]
+ *  - {Comparing}[#class-String-label-Methods+for+Comparing]
+ *  - {Modifying +self+}[#class-String-label-Methods+for+Modifying+self]
+ *  - {Converting to New String}[#class-String-label-Methods+for+Converting+to+New+String]
+ *  - {Converting to Non-String}[#class-String-label-Methods+for+Converting+to+Non--5CString]
+ *  - {Iterating}[#class-String-label-Methods+for+Iterating]
+ *
+ *  === Methods for Creating a \String
+ *
+ *  - ::new:: Returns a new string.
+ *  - ::try_convert:: Returns a new string created from a given object.
+ *
+ *  === Methods for a Frozen/Unfrozen String
+ *
+ *  - {#+string}[#method-i-2B-40]:: Returns a string that is not frozen:
+ *                                  +self+, if not frozen; +self.dup+ otherwise.
+ *  - {#-string}[#method-i-2D-40]:: Returns a string that is frozen:
+ *                                  +self+, if already frozen; +self.freeze+ otherwise.
+ *  - #freeze:: Freezes +self+, if not already frozen; returns +self+.
+ *
+ *  === Methods for Querying
+ *
+ *  _Indexes_
+ *
+ *  - {#=~}[#method-i-3D~]:: Returns the index of the first substring that matches a given Regexp or other object;
+ *                           returns +nil+ if no match is found.
+ *  - #index:: Returns the index of the first occurrence of a given substring;
+ *             returns +nil+ if none found.
+ *  - #rindex:: Returns the index of the _last_ occurrence of a given substring;
```

Comment:
Seems odd to emphasize `last` here without emphasizing `first` for `index`.

Reply:
Fixed.